sw/device/tests/aes_prng_reseed_test.c

```diff
@@ -0,0 +1,243 @@
+
+  // Generate key with index 0
+  generate_new_key(&key, 0);
+
+  // Prepare transaction
+  dif_aes_transaction_t transaction = {
+      .operation = kDifAesOperationEncrypt,
+      .mode = kDifAesModeEcb,
+      .key_len = kDifAesKey128,
+      .key_provider = kDifAesKeySoftwareProvided,
+      .mask_reseeding = kDifAesReseedPer64Block,
+      .manual_operation = kDifAesManualOperationAuto,
+      .reseed_on_key_change = true,
+      .force_masks = false,
+      .ctrl_aux_lock = false,
+  };
+  // Start the AES operation
+  // Wait for AES to be idle before starting encryption
+  aes_idle = false;
+  do {
+    CHECK_DIF_OK(dif_aes_get_status(&aes, kDifAesStatusIdle, &aes_idle));
+  } while (!aes_idle);
+
+  CHECK_DIF_OK(dif_aes_start(&aes, &transaction, &key, NULL));
+
+  dif_aes_data_t plain_text[kAesNumBlocks];
+  dif_aes_data_t cipher_text[kAesNumBlocks];
+
+  // Initialize plaintext data dynamically
+  // Create plaintext with random data
+  dif_rv_core_ibex_t rv_core_ibex;
+  CHECK_DIF_OK(dif_rv_core_ibex_init(
+      mmio_region_from_addr(TOP_EARLGREY_RV_CORE_IBEX_CFG_BASE_ADDR),
+      &rv_core_ibex));
+  for (int i = 0; i < kAesNumBlocks; ++i) {
+    for (int j = 0; j < kDifAesBlockNumBytes; ++j) {
+      uint32_t rand;
+      TRY(rv_core_ibex_testutils_get_rnd_data(&rv_core_ibex, 2000, &rand));
+      plain_text[i].data[j] = rand;
+    }
+  }
+
+  // Process the first two blocks using dif_aes_process_data()
+  dif_aes_data_t aes_block_plain_text_12[2];
+  dif_aes_data_t aes_block_cipher_text_12[2];
+
+  // Collect the first two blocks
+  aes_block_plain_text_12[0] = plain_text[0];
+  aes_block_plain_text_12[1] = plain_text[1];
+
+  // Encrypt the first two blocks
+  LOG_INFO("Using dif_aes_process_data() for the first two blocks");
+  CHECK_DIF_OK(dif_aes_process_data(&aes, aes_block_plain_text_12,
+                                    aes_block_cipher_text_12, 2));
+
+  // Store the encrypted data
+  cipher_text[0] = aes_block_cipher_text_12[0];
+  cipher_text[1] = aes_block_cipher_text_12[1];
+
+  // Now start encryption process for the remaining blocks starting from i = 2
+  // Start encryption process
+  for (int i = 2; i < kAesNumBlocks; ++i) {
+    // Wait for input ready
+    input_ready = false;
+    do {
+      CHECK_DIF_OK(
+          dif_aes_get_status(&aes, kDifAesStatusInputReady, &input_ready));
+    } while (!input_ready);
+
+    // Load data
+    CHECK_DIF_OK(dif_aes_load_data(&aes, plain_text[i]));
+
+    // Check for stall after entropy disabled
+    if (i == disable_entropy_after_block) {
+      output_valid = false;
+      CHECK_DIF_OK(
+          dif_aes_get_status(&aes, kDifAesStatusOutputValid, &output_valid));
+      if (output_valid) {
+        LOG_INFO(
+            "AES encryption has output_valid as expected after entropy "
+            "disabled at block %d",
+            i);
+      } else {
+        LOG_ERROR("ERROR:AES encryption did not generate output_valid");
+      }
```

Comment:
```suggestion
      CHECK(output_valid, "ERROR:AES encryption did not generate output_valid");
```
Please change this. With this `CHECK`, the test fails (and prints the according `FAIL` message) when `output_valid` is not true. Same for the `aes_force_prng_reseed` test.